pymeasure/instruments/rohdeschwarz/sfm.py

```diff
@@ -0,0 +1,1506 @@
+        )
+
+    serial_flowcontrol = Instrument.control(
+        "SYST:COMM:SER:PACE?",
+        "SYST:COMM:SER:PACE %s",
+        """ A string property that controls the serial handshake type used in serial communication
+
+        Possible values are:
+
+        ======  =======
+        Value   Meaning
+        ======  =======
+        NONE    no flow-control/handshake
+        XON     XON/XOFF flow-control
+        ACK     hardware handshake with RTS&CTS
+        ======  =======
+
+        """,
+        validator = strict_discrete_set,
+        values=["NONE","XON","ACK"],
+        )
+
+    serial_parity = Instrument.control(
+        "SYST:COMM:SER:PAR?",
+        "SYST:COMM:SER:PAR %s",
+        """ A string property that controls the parity type used for serial communication
+
+        Possible values are:
+
+        ======  =======
+        Value   Meaning
+        ======  =======
+        NONE    no parity
+        EVEN    even parity
+        ODD     odd parity
+        ONE     parity bit fixed to 1
+        ZERO    parity bit fixed to 0
+        ======  =======
+
+        """,
+        validator = strict_discrete_set,
+        values=["NONE","EVEN","ODD","ONE","ZERO"],
+        )
+
+    serial_stopbits = Instrument.control(
+        "SYST:COMM:SER:SBIT?",
+        "SYST:COMM:SER:SBIT %g",
+        """ A int property that controls the number of stop-bits used in serial communication,
+
+        Possible values are: 1 or 2
+        """,
+        validator = strict_discrete_set,
+        values=[1,2],
+        )
+
+    date = Instrument.measurement(
+        "SYST:DATE?",
+        """
+        A list property for the date of the RTC in the unit
+
+        """,
+        )
+
+    time = Instrument.measurement(
+        "SYST:TIME?",
+        """
+        A list property for the time of the RTC in the unit
+
+        """,
+        )
+
+    basic_info = Instrument.measurement(
+        "SYST:INF:BAS?",
+        """
+        A String property containing infomation about the hardware modules installed in the unit
+
+        """,
+        )
+
+    #TODO: SUBS --> SUB[1..6]
+    subsystem_info = Instrument.measurement(
+        "SYST:INF:SUBS?",
+        """
+        A String property containing infomation about the system configuration
+        """,
+)
+
+    #Unit subsystem (3.6.9)
+    scale_volt = Instrument.control(
+        "UNIT:VOLT?",
+        "UNIT:VOLT %s",
+        """ A string property that controls the unit to be used for voltage entries on the unit
+
+        Possible values are:
+        AV,FV, PV, NV, UV, MV, V, KV, MAV, GV, TV, PEV, EV,
+        DBAV, DBFV, DBPV, DBNV, DBUV, DBMV, DBV, DBKV, DBMAv, DBGV, DBTV, DBPEv, DBEV
+        """,
+        validator = strict_discrete_set,
+        values= ["AV","FV", "PV", "NV", "UV", "MV", "V", "KV", "MAV", "GV",
+                 "TV", "PEV", "EV", "DBAV", "DBFV", "DBPV", "DBNV", "DBUV",
+                 "DBMV", "DBV", "DBKV", "DBMAv", "DBGV", "DBTV", "DBPEv", "DBEV"],
+        )
```

Comment:
Wow, quite long and exhaustive list of properties!!! Indeed a lot of work, just minor comment, I see a some TODOs in the code. Is this something you want to address now or is left as reminder for the future modifications ?

Reply:
1. I went basically trough the manual and implentened all the properties, these kind of units have many different options. after a while it becomes a copy & paste job ;-)
2. And many properties have big list of possible values, this is why I often went to the longer docstrings with better formatting.


3. The TODOs I would like to leave for the future, as for my use-case I do not see much use of having the different systems (=configurations) setup in mine.
Unfortunately R&S way of implementing this system switching with the numbers added into different command-words does not make iit easy to implement in pymeasure.
It could be done with different classes, but 3 or 4 different classes would make the implmentation more difficult then it is at the moment.